src/simsopt/mhd/bootstrap.py

```diff
@@ -0,0 +1,740 @@
+                return lambd / (np.mean(np.sqrt(1 - lambd * modB[:, js]) * sqrtg[:, js]) \
+                                / (twopi * dVds[js]))
+
+            integral = quad(integrand, 0, 1 / modBmax)
+            f_t[js] = 1 - 0.75 * fsa_B2[js] * integral[0]
+
+    else:
+        raise ValueError('Input arrays must be 2D or 3D')
+
+    logging.debug(f'Bmin: {Bmin}  Bmax: {Bmax}  epsilon: {epsilon}  '
+                  f'fsa_B2: {fsa_B2}  fsa_1overB: {fsa_1overB}  f_t: {f_t}')
+    return Bmin, Bmax, epsilon, fsa_B2, fsa_1overB, f_t
+
+
+def j_dot_B_Redl_fits(s, ne, Te, Ti, Zeff, G, R, iota, epsilon, f_t, psi_edge, helicity_N):
+    r"""
+    Compute the bootstrap current (specifically
+    :math:`\left<\vec{J}\cdot\vec{B}\right>`) using the formulae in
+    Redl et al, Physics of Plasmas 28, 022502 (2021).
+
+    The profiles of ne, Te, Ti, and Zeff should all be instances of
+    subclasses of :obj:`simsopt.mhd.profiles.Profile`, i.e. they should
+    have ``__call__()`` and ``dfds()`` functions. If ``Zeff == None``, a
+    constant 1 is assumed. If ``Zeff`` is a float, a constant profile will
+    be assumed.
+
+    ``ne`` should have units of 1/m^3. ``Ti`` and ``Te`` should have
+    units of eV.
+
+    The input variable ``s`` is a 1D array of values of normalized
+    toroidal flux.  The input arrays ``G``, ``R``, ``iota``,
+    ``epsilon``, and ``f_t``, should be 1d arrays evaluated on this
+    same ``s`` grid. The bootstrap current
+    :math:`\left<\vec{J}\cdot\vec{B}\right>` will be computed on this
+    same set of flux surfaces.
+
+    Args:
+        s: A 1D array of values of normalized toroidal flux.
+        ne: A :obj:`~simsopt.mhd.profiles.Profile` object with the electron density profile.
+        Te: A :obj:`~simsopt.mhd.profiles.Profile` object with the electron temperature profile.
+        Ti: A :obj:`~simsopt.mhd.profiles.Profile` object with the ion temperature profile.
+        Zeff: A :obj:`~simsopt.mhd.profiles.Profile` object with the profile of the average
+            impurity charge :math:`Z_{eff}`. Or, a single number can be provided if this profile is constant.
+            Or, if ``None``, Zeff = 1 will be used.
+        G: A 1D array with the flux function multiplying :math:`\nabla\varphi` in the Boozer covariant representation,
+            equivalent to :math:`R B_{toroidal}` in axisymmetry.
+        R: A 1D array with the effective major radius to use when evaluating
+            the collisionality in the Sauter/Redl formulae.
+        iota: A 1D array with the rotational transform.
+        epsilon: A 1D array with the effective inverse aspect ratio to use for
+            evaluating the collisionality in the Sauter/Redl formulae.
+        f_t: A 1D array with the effective trapped fraction.
+        psi_edge: The toroidal flux (in Webers) divided by (2pi) at the boundary s=1
+        helicity_N: 0 for quasi-axisymmetry, or +/- the number of field periods for quasi-helical symmetry.
```

Comment:
It might be helpful to use the same naming convention here for helicity_N and in boozer.Quasisymmetry for m and n.

Reply:
Changed `helicity_N` (which is +/- nfp for QH) to `helicity_n`, which is +/- 1 for QH, so it is now identical to the parameter `helicity_n` in `simsopt.mhd.vmec_diagnostics.QuasisymmetryRatioResidual`.

Reply:
Now helicity_n is used consistently in j_dot_B_Redl, QuasisymmetryRatioResidual, and Quasisymmetry.